### PR TITLE
test: Add additional contains expression tests

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Follow-up to #2991.

## Rationale for this change

The optimized `contains` implementation merged in #2991 had good test coverage but was missing a few edge cases.

## What changes are included in this PR?

Adds integration tests for:
- Null haystack (`contains(NULL, 'pattern')`) to verify null propagation
- Case sensitivity (`contains(name, 'james')` vs `'James'`)
- Both-columns path (`contains(col1, col2)`) exercising the array-array code path, including null in either position

## How are these changes tested?

All new tests use `checkSparkAnswerAndOperator` to verify both correctness against Spark and that Comet's native operator is used.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala [hunk 1]
```diff
@@ -1181,6 +1181,21 @@ class CometExpressionSuite extends CometTestBase with AdaptiveSparkPlanHelper {
...
+      // Test with null haystack
+      sql(s"insert into $table values(6, null)")
+      checkSparkAnswerAndOperator(sql(s"select id, contains(name, 'Rose') from $table"))
+
+      // Test case sensitivity (should not match)
+      checkSparkAnswerAndOperator(sql(s"select id from $table where contains(name, 'james')"))
+    }
+  }
+
+  test("contains with both columns") {
```
Comment (comphead) [inline spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala:1194]: are you planning also move it to `sql` tests?

**Review by comphead — APPROVED** [review]
Thanks @andygrove

File: spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala (hunk 1 above)
Comment (andygrove) [reply spark/src/test/scala/org/apache/comet/CometExpressionSuite.scala:1194]: maybe .. we have many, many tests to move over to the new approach. I guess we need to figure out a plan.
